UAGimbalAdapter.m

```diff
@@ -75,42 +77,42 @@ - (void)setBluetoothPoweredOffAlertEnabled:(BOOL)bluetoothPoweredOffAlertEnabled
 }
 
 - (void)startWithGimbalAPIKey:(NSString *)gimbalAPIKey {
-    if (self.isStarted) {
-        return;
-    }
-
-    if (gimbalAPIKey.length) {
-        [Gimbal setAPIKey:gimbalAPIKey options:nil];
-    } else if (![GMBLPlaceManager isMonitoring]) {
-        NSLog(@"GMBLPlaceManager is not previously started and API key is not provided. Unable to start Gimbal Adapter.");
-        return;
-    }
-
-    self.placeManager.delegate = self;
-    [GMBLPlaceManager startMonitoring];
+    [Gimbal setAPIKey:gimbalAPIKey options:nil];
+    [Gimbal start];
     self.started = YES;
+    self.placeManager.delegate = self;
+    [self setDeviceAttributes];
     UA_LDEBUG(@"Started Gimbal Adapter. Gimbal application instance identifier: %@", [Gimbal applicationInstanceIdentifier]);
 }
 
-- (void)stop {
-    if (!self.isStarted) {
-        return;
+- (void)setDeviceAttributes {
+    NSMutableDictionary *deviceAttributes = [NSMutableDictionary new];
+    if ([[self.deviceAttributesManager getDeviceAttributes] count] > 0) {
+        [deviceAttributes addEntriesFromDictionary:[self.deviceAttributesManager getDeviceAttributes]];
+    }
+    if ([UAirship namedUser].identifier) {
+        [deviceAttributes setObject:[UAirship namedUser].identifier forKey:@"ua.nameduser.id"];
```

Comment:
If the developer never sets the nameduser this will error out:

2017-05-24 13:24:33.334893 Sample[28335:8301375] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '*** setObjectForKey: object cannot be nil (key: ua.nameduser.id)'
*** First throw call stack:
(0x1903c51b8 0x18edfc55c 0x1902a5a0c 0x10007962c 0x1000793a8 0x1000745c0 0x1962ec6a4 0x1964fca98 0x196502808 0x196517104 0x1964ff7ec 0x191f9b92c 0x191f9b798 0x191f9bb40 0x190372b5c 0x1903724a4 0x1903700a4 0x19029e2b8 0x1962e57b0 0x1962e0534 0x100077e4c 0x18f2815b8)
libc++abi.dylib: terminating with uncaught exception of type NSException